cirq-google/cirq_google/engine/engine_client.py

```diff
@@ -434,17 +418,21 @@ async def create_job_async(
 
         Raises:
             ValueError: If the priority is not between 0 and 1000.
-            ValueError: If neither `processor_id` or `processor_ids` are set.
             ValueError: If  only one of `run_name` and `device_config_name` are specified.
-            ValueError: If `processor_ids` has more than one processor id.
             ValueError: If either `run_name` and `device_config_name` are set but
                 `processor_id` is empty.
         """
         # Check program to run and program parameters.
         if priority and not 0 <= priority < 1000:
             raise ValueError('priority must be between 0 and 1000')
+
         if not processor_id:
-            raise ValueError('Must specify a processor id when creating a job.')
+            if run_name or device_config_name:
```

Comment:
I think we can omit this condition and just rely on the `else` case: if there's no processor_id, then make them specify it regardless of the `run_name` or `device_config_name` status.